app/upload/views.py

```diff
@@ -25,6 +25,13 @@ def upload_document(service_id):
         ), 400
     file_content = request.files['document'].read()
 
+    filename = request.form.get('filename')
+    # If the API passed None as an argument, it could be converted
+    # as a string value `None`. Convert it back to a proper None.
+    # See https://bugs.python.org/issue18857
+    if filename and filename.lower() == 'none':
```

Comment:
This is an oddball, but if we allow 'None'/'none' etc. as a filename in the API, we'll swallow it up. Maybe they deserve that? But I could see a client API implementation being thrown for a loop if they mis-configure something.

Reply:
Added a test on the API PR to prove that the `filename` argument is not passed to the document API is the value is `None` https://github.com/cds-snc/notification-api/pull/1249/commits/bd956c48fa5279121fb5892e83b92c4435c9cb75

It was not needed to change the logic, `requests` already removes `None` keys before making POST requests. Evidence: https://github.com/psf/requests/blob/913880c45a3a8c3bf6b298e9c38709cd95a9c97c/requests/models.py#L83-L102

Reply:
I was actually thinking of some odd situation where filename is actually str('None') from the user / notify client. I've seen cases where systems improperly store None as str('None') in the db. Again I know it's a total edge case.